test/unit/helpers/QtiPackageExporterTest.php

```diff
@@ -0,0 +1,144 @@
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->exporterMock = $this->createMock(taoQtiTest_models_classes_export_TestExport22::class);
+        $this->fileSystemServiceMock = $this->createMock(FileSystemService::class);
+        $this->fileHelperServiceMock = $this->createMock(FileHelperService::class);
+        $this->fileHelperServiceMock->method('createTempDir')->willReturn('FAKE_TMP_DIR');
+
+        $this->subject = new QtiPackageExporter(
+            $this->exporterMock,
+            $this->fileSystemServiceMock,
+            $this->fileHelperServiceMock
+        );
+    }
+
+    public function testExportDeliveryQtiPackage_ThrowsExceptionWhenExportFails(): void
+    {
+        $testUri = 'FAKE_TEST_URI';
+
+        $expectedReport = common_report_Report::createFailure('FAKE ERROR MESSAGE');
+        $this->exporterMock->method('export')
+            ->willReturn($expectedReport);
+
+        $this->expectException(common_Exception::class);
+        $this->subject->exportDeliveryQtiPackage($testUri);
+    }
+
+    /**
+     * @param array $reportData
+     *
+     * @dataProvider dataProviderReportDataWithoutValidPath
+     */
+    public function testExportQtiTestPackageToFile_ThrowsExceptionWhenReportDoesNotHaveValidPath(array $reportData): void
+    {
+        $testUri = 'FAKE_TEST_URI';
+        $fileSystemId = 'FILE_SYSTEM_ID';
+        $filePath = 'FILE_PATH';
+
+        $expectedReport = common_report_Report::createSuccess('FAKE ERROR MESSAGE');
+        $expectedReport->setData($reportData);
+
+        $this->exporterMock->method('export')
+            ->willReturn($expectedReport);
+
+        $this->expectException(common_Exception::class);
+        $this->subject->exportQtiTestPackageToFile($testUri, $fileSystemId, $filePath);
+    }
+
+    public function testExportQtiTestPackageToFile_ReturnsValidFileAfterSuccessfulExport(): void
+    {
+        $testUri = 'FAKE_TEST_URI';
+        $fileSystemId = 'FILE_SYSTEM_ID';
+        $filePath = 'FILE_PATH';
+
+        $expectedReport = common_report_Report::createSuccess('FAKE ERROR MESSAGE');
+        $expectedReport->setData(['path' => 'FAKE_QTI_PACKAGE_PATH']);
+        $this->exporterMock->method('export')
+            ->willReturn($expectedReport);
+
+        $this->fileHelperServiceMock->method('readFile')
+            ->willReturn('FILE_RESOURCE');
+
+        $fileMock = $this->createMock(File::class);
+        $directoryMock = $this->createMock(Directory::class);
+        $directoryMock->method('getFile')
+            ->willReturn($fileMock);
+
+        $this->fileSystemServiceMock->method('getDirectory')
+            ->willReturn($directoryMock);
+
+        $result = $this->subject->exportQtiTestPackageToFile($testUri, $fileSystemId, $filePath);
+
+        self::assertInstanceOf(
+            File::class,
+            $result,
+            'Method must return instance of File in case of successful export.'
+        );
```

Comment:
You are not verifying if what you receive is a valid file or not or if it is the file what you would expect. This whole test case can be omitted just by adding return type hint to the `exportQtiTestPackageToFile` method.

Reply:
There is already return type declaration. I had the same feeling about this test case. Removed.